applications/crossbar/src/modules/cb_tasks.erl

```diff
@@ -0,0 +1,433 @@
+%% @public
+%% @doc
+%% If the HTTP verb is PATCH, execute the actual action, usually a db save
+%% (after a merge perhaps).
+%% @end
+%%--------------------------------------------------------------------
+-spec patch(cb_context:context(), path_token()) -> cb_context:context().
+patch(Context, TaskId) ->
+    case kz_tasks:start(TaskId) of
+        {'ok', Task} -> crossbar_util:response(Task, Context);
+        {'error', 'no_categories'} ->
+            no_categories(Context, TaskId);
+        {'error', 'already_started'} ->
+            Msg = kz_json:from_list([{<<"reason">>, <<"task already started">>}
+                                    ,{<<"cause">>, TaskId}
+                                    ]),
+            cb_context:add_system_error('bad_identifier', Msg, Context)
+    end.
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% If the HTTP verb is DELETE, execute the actual action, usually a db delete
+%% @end
+%%--------------------------------------------------------------------
+-spec delete(cb_context:context(), path_token()) -> cb_context:context().
+delete(Context, TaskId) ->
+    case kz_tasks:remove(TaskId) of
+        {'ok', Task} -> crossbar_util:response(Task, Context);
+        {'error', 'task_running'} ->
+            Msg = kz_json:from_list([{<<"message">>, <<"task is running">>}
+                                    ,{<<"cause">>, TaskId}
+                                    ]),
+            cb_context:add_system_error('bad_identifier', Msg, Context)
+    end.
+
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+%% @private
+-spec set_db(cb_context:context()) -> cb_context:context().
+set_db(Context) ->
+    cb_context:set_account_db(Context, ?KZ_TASKS_DB).
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Load an instance from the database
+%% @end
+%%--------------------------------------------------------------------
+-spec read(ne_binary(), cb_context:context()) -> cb_context:context().
+read(TaskId, Context) ->
+    case kz_tasks:read(TaskId) of
+        {'error', 'not_found'} ->
+            crossbar_util:response_bad_identifier(TaskId, Context);
+        {'ok', TaskJObj} ->
+            cb_context:setters(Context, [{fun cb_context:set_resp_status/2, 'success'}
+                                        ,{fun cb_context:set_resp_data/2, TaskJObj}
+                                        ])
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Attempt to load a summarized listing of all instances of this
+%% resource.
+%% @end
+%%--------------------------------------------------------------------
+-spec summary(cb_context:context()) -> cb_context:context().
+summary(Context) ->
+    ViewOptions = [{'key', cb_context:account_id(Context), 'undefined'}
```

Comment:
@fenollp is this is a compound key, it should be a list
`ViewOptions = [{'key', [cb_context:account_id(Context), 'undefined']}`

but 'undefined' doesn't seem a proper argument to key ?